src/commands/shell.rs

```diff
@@ -73,20 +91,178 @@ pub async fn command(args: Args, _json: bool) -> Result<()> {
         eprintln!("No service linked, skipping service variables");
     }
 
+    // disable dead code warning for windows_shell_detection
+    #[allow(dead_code)]
+    enum WindowsShell {
```

Comment:
Can this and all shell detection code be extracted outside of the `command` function

Reply:
Will do